app/templates/base.html.j2

```diff
@@ -36,9 +38,32 @@
 
         {# Date/Time formatting #}
         {{ moment.include_moment() }}
-        {# Bootstrap 5 JavaScript #}
+
+        <!-- Bootstrap 5 JavaScript -->
         <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-C6RzsynM9kWDrMNeT87bh95OGNyZPhcTNXj1NW7RuBCsyN/o0jlpcV8Qyq46cDfL" crossorigin="anonymous"></script>
-      </div>
-    </div>
+
+        {# Add CSRF token to each AJAX request. Source:  https://flask-wtf.readthedocs.io/en/1.2.x/csrf/ #}
+        <script type="text/javascript">
+            var csrf_token = "{{ csrf_token() }}";
+            $.ajaxSetup({
+                beforeSend: function(xhr, settings) {
+                    if (!/^(GET|HEAD|OPTIONS|TRACE)$/i.test(settings.type) && !this.crossDomain) {
+                        xhr.setRequestHeader("X-CSRFToken", csrf_token);
+                    }
+                }
+            });
+        </script>
```

Comment:
Did you add this into branch?  It already exists in main.  It looks like you just copied and pasted it in, but I could be wrong.

Reply:
It's from the previous branch I did, New-Customer-Modal-Form, I created this branch from it after I submitted the PR for New-Customer-Modal-Form. So What happened was on Saturday I went to rebase this branch but I decided to try something new and used the GitLens interactive editor in VSCode. I must've done something wrong because it duplicated all my commits, the code was fine though but I ended up undoing the commits. I chose to rebase with main but it looks like it rebased with New-Customer-Modal-Form.
![Commit Graph](https://github.com/Wcarey7/KillawattElectrical/assets/13939125/66f37a42-6dbf-4db9-b786-4a2274add9d9)

Reply:
Got it.  When you rebase right before you merge the PR, are you marking most of the commits as "squash"? Or are you "picking" all of them?

Reply:
I only selected a couple to squash.

Reply:
We don't want all of these commits added to the history. It just clusters it up. For example, you have more than one favicon commit and multiple commits related to the customer modal. Consolidate those and any others you see into one commit. You can use reword to change the commit message to something more descriptive.

Do you want me to show you what I mean?

Reply:
Yeah, how do I reword it? I cant find an option to do it. The commits didn't get squashed correctly like I thought they would when I attempted the rebase.

Reply:
Let's go into it tonight. I don't want to do it too late. Call me between 5 and 6. I'll be unavailable between 3 and 5, but should be back home by 5 or so.